### PR TITLE
fix: only call activedescendant focus when combobox/dropdown is opened

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

Related to #31140 and #31415

Moving the activedescendant `.focus()` logic to Listbox's initial render also caused the page to attempt to scroll the active option into view before it was visible on the page (since the listbox is rendered on focus for combobox/dropdown).

Usually this doesn't do anything, but if the scrollParent isn't `null`, it can scroll the page (or closest scroll parent).

## New Behavior

Now Listbox only calls the initial `activeDescendantController.focus()` logic if it does not have a parent context (so if rendered inline in the page, for example). Combobox & Dropdown call it when the listbox is opened, which is what happened previously.

## Related Issue(s)

Fixes a Teams issue raised in chat.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `103.623 kB`<br />`34.007 kB` | `103.763 kB`<br />`34.054 kB` | `140 B` <br />`47 B` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `104.209 kB`<br />`33.914 kB` | `104.349 kB`<br />`33.963 kB` | `140 B` <br />`49 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.094 MB`<br />`270.233 kB` | `1.094 MB`<br />`270.273 kB` | `140 B` <br />`40 B` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `181.227 kB`<br />`54.489 kB` | `181.367 kB`<br />`54.538 kB` | `140 B` <br />`49 B` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `106.611 kB`<br />`35.546 kB` | `106.751 kB`<br />`35.596 kB` | `140 B` <br />`50 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.326 kB`<br />`63.005 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/cf68d198b62eeccb0031517a4558f73877ff92f1'>cf68d198b62eeccb0031517a4558f73877ff92f1</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Avatar-mount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Avatar-mount.html">632</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Button-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Button-mount.html">301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Field-mount_base.html">1129</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Field-mount.html">1081</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProvider-mount_base.html">716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProvider-mount.html">710</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProviderWithTheme-mount_base.html">87</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProviderWithTheme-mount.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">34</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/MakeStyles-mount_base.html">878</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/MakeStyles-mount.html">865</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Persona-mount_base.html">1772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/Persona-mount.html">1712</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/SpinButton-mount_base.html">1432</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/SpinButton-mount.html">1379</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/SwatchPicker-mount_base.html">1680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32156/perf-test-react-components/SwatchPicker-mount.html">1651</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-combobox-d70f6062-309a-423a-9657-e1b8621c8b27.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-combobox-d70f6062-309a-423a-9657-e1b8621c8b27.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32156&commitId=0f3b15289b7c0bd2ef8ff033836bdeba0daee5d6&env=prod&clientType=FLUENTUI)


  <details><summary><b>SwatchPicker Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.default - High Contrast.default.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (smhigley) [reply change/@fluentui-react-combobox-d70f6062-309a-423a-9657-e1b8621c8b27.json:1]: definitely not from this PR 😅

**Pull request merged**